spec/asyncapi.md

```diff
@@ -848,7 +851,18 @@ This object MAY be extended with [Specification Extensions](#specificationExtens
   },
   "traits": [
     { "$ref": "#/components/operationTraits/kafka" }
-  ]
+  ],
+  "messages": [
+    { "$ref": "/components/messages/userSignedUp" }
+  ],
```

Comment:
What i rember is the sentence:
why having an array if you never reference the key.

Reply:
Yeah, I think it was the opposite. It's better to reference messages in a map by their id than to do it by their number in an array. So it should stay as a map, like in `channels`. That said, it should be a map of `$ref` objects pointing to Message Objects instead of a map of Message Objects. Just to prevent duplication. So we should change the description above to say it's "a map" instead of "a list".

Reply:
But we never use the map key or array index. Because we agreed to use $ref

Reply:
Oh, true! We use the key on `channels` but not the one that we would define here. Yeah, in such a case I think it makes sense it stays as an array. You're right.